mempool/src/pool/store.rs

```diff
@@ -123,6 +123,32 @@ impl MempoolStore {
         self.txs_by_id.get(&id.get())
     }
 
+    pub(super) fn assert_valid(&self) {
+        #[cfg(test)]
+        self.assert_valid_inner()
+    }
+
+    #[cfg(test)]
+    pub(super) fn assert_valid_inner(&self) {
+        let entries = self.txs_by_descendant_score.values().flatten().collect::<Vec<_>>();
+        for id in self.txs_by_id.keys() {
+            assert_eq!(
+                entries.iter().filter(|entry_id| ***entry_id == Id::new(*id)).count(),
+                1
+            )
+        }
+        for entry in self.txs_by_id.values() {
+            for child in &entry.children {
+                assert!(self
+                    .txs_by_id
+                    .get(&child.get())
+                    .expect("child")
+                    .parents
+                    .contains(&entry.tx_id()))
+            }
+        }
```

Comment:
Cool stuff. Do you plan to extend this further? I guess there are still more consistency invariants not covered by these checks.

Reply:
There's plenty more than can be checked I'm sure, haven't given it much thought yet but at least wanted the two invariants that you mentioned in the original PR to be covered :)
If anything else comes to mind feel free to suggest and we'll add it.
For example, a similar check as was done for txs_by_descendant_score can be done for txs_by_creation_time.